plugins/coverbrowser.koplugin/covermenu.lua

```diff
@@ -232,8 +232,7 @@ function CoverMenu:updateItems(select_number)
     -- we replace it by ours.
     -- (FileManager may replace file_chooser.showFileDialog after we've been called once, so we need
     -- to replace it again if it is not ours)
-    if not self.showFileDialog_ours -- never replaced
-            or self.showFileDialog ~= self.showFileDialog_ours then -- it is no more ours
+    if self.showFileDialog and self.showFileDialog ~= self.showFileDialog_ours then
         -- We need to do it at nextTick, once FileManager has instantiated
         -- its FileChooser completely
```

Comment:
(Not really want to re-read all my old replacement logic) but why this change ? Simplification/optimisation (but may be a little less obvious to see the logic), or needed change because of all the other changes? Can self.showFileDialog not exist?

Reply:
It tried to patch `showFileDialog` when showing History and Collection.

Reply:
> Can self.showFileDialog not exist?

Yes, when showing History. And
https://github.com/koreader/koreader/blob/b832d43d985778e36841b7d08cf3db5436f26249/plugins/coverbrowser.koplugin/main.lua#L646

Reply:
> It tried to patch showFileDialog when showing History and Collection.

Sorry for being dense, but you are becoming as precise as @Frenzie (that is: not a lot :)) when answeing comments/questions :)

> but why this change ?

A Simplification/optimisation
B Needed change because of the other changes
C Unrelated bug you are solving

Reply:
`CoverMenu.updateItems` is a common patch for FM, Hist and Coll.
But in Hist and Coll `showFileDialog` does not exist.

Reply:
Re-sorry for being extra dense.

**A B or C ?**

I'm guessing (after a few minutes reading your comments, that you could have spared me :)) it is a mix of A and C: unrelated to everything else in this PR, and we were (when in Hist and Coll) replacing (uselessly) that method that was never used? Right?

Reply:
> we were (when in Hist and Coll) replacing (uselessly) that method that was never used? Right?

We tried to replace the method that did not exist.